src/core/gltf2importer/bufferaccessorparser.cpp

```diff
@@ -106,6 +110,60 @@ Qt3DRender::QAttribute::VertexBaseType accessorTypeFromJSON(int componentType)
         return Qt3DRender::QAttribute::Float;
     }
 }
+
+template<class IndexType>
+bool copySparseValues(int count, const QByteArray &indicesBuffer, const QByteArray &valuesBuffer,
+                      QByteArray &targetBuffer, int attributeOffset, int stride,
+                      int valueSize)
+{
+    if (stride == 0)
+        stride = valueSize;
+    const IndexType *indices = reinterpret_cast<const IndexType *>(indicesBuffer.data());
+    for (int i = 0; i < count; ++i) {
+        IndexType ndx = indices[i];
+        char *target = targetBuffer.data() + stride * ndx + attributeOffset;
+        const char *source = valuesBuffer.data() + i * valueSize;
+        memcpy(target, source, static_cast<size_t>(valueSize));
+    }
+    return true;
+}
+
+template<class IndexType>
+bool copySparseValues(int count, const QByteArray &indicesBuffer, const QByteArray &valuesBuffer,
+                      QByteArray &targetBuffer, int attributeOffset, int stride,
+                      Qt3DRender::QAttribute::VertexBaseType componentType, int numComponents)
+{
+    switch (componentType) {
```

Comment:
wouldn't it be easier to read if in the switch we only computed the valued size allowing to then only have a single call to copySparseValues?

Reply:
good call